internal/tools/orchestrator/hepa/providers/domain/domain.service.go

```diff
@@ -117,9 +122,63 @@ func (s *domainService) ChangeInnerIngress(ctx context.Context, req *pb.ChangeIn
 	}
 	return
 }
+
+func (s *domainService) preCheckPermission(userID string, services *bundle.GetRuntimeServicesResponseData) error {
```

Comment:
Attempting to implement authentication at the middleware level. [code](https://github.com/erda-project/erda/blob/3d410d4ef057d09ede67ea40bf7ee8ce16648d2d/internal/tools/orchestrator/hepa/providers/domain/provider.go#L63)